### PR TITLE
feat: add elb logtanks

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       70.796s
```

### CODE REVIEW

File: docs/resources/elb_logtanks.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+subcategory: "Dedicated Load Balance (Dedicated ELB)"
+---
+
+# huaweicloud_elb_logtank
+
+Manage a ELB logtank resource within HuaweiCloud.
+
+## Example Usage
+
+### create with system template
```
Comment (hmbig2) [inline docs/resources/elb_logtanks.md]: s/create/Create

File: docs/resources/elb_logtanks.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+```
+
+### create with custom template
+
+```hcl
+variable "loadbalancer_id" {}
+variable "group_id" {}
+variable "topic_id" {}
+resource "huaweicloud_elb_logtank" "test" {
```
Comment (hmbig2) [inline docs/resources/elb_logtanks.md]: this config is same with `create with system template`.   lost custom argument?

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+	logTank, err := logtanks.Get(elbClient, d.Id()).Extract()
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "logtanks")
+	}
+
+	log.Printf("[DEBUG] Retrieved logtank %s: %#v", d.Id(), logTank)
+
+	mErr := multierror.Append(nil,
+		d.Set("loadbalancer_id", logTank.LoadbalancerID),
```
Comment (hmbig2) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: add region set

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"loadbalancer_id": {
+				Type:     schema.TypeString,
```
Comment (hmbig2) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: if this argument cant  be update, please set to forcenew

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&logTanks,
+		getELBLogTankResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckElbV3LogTanksDestroy,
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: s/testAccCheckElbV3LogTanksDestroy/rc.checkResourceDestroy

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckElbV3LogTanksDestroy(s *terraform.State) error {
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: ditto

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+func getELBLogTankResourceFunc(c *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := c.ElbV3Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmtp.Errorf("Error creating HuaweiCloud elb client: %s", err)
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: s/fmtp/fmt

File: docs/resources/elb_logtanks.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
+---
+subcategory: "Dedicated Load Balance (Dedicated ELB)"
+---
+
+# huaweicloud_elb_logtank
+
+Manage a ELB logtank resource within HuaweiCloud.
```
Comment (Lance52259) [inline docs/resources/elb_logtanks.md]: s/a/an

File: docs/resources/elb_logtanks.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+```hcl
+variable "loadbalancer_id" {}
+variable "group_id" {}
+variable "topic_id" {}
+resource "huaweicloud_elb_logtank" "test" {
+  loadbalancer_id = var.loadbalancer_id
+  log_group_id    = var.group_id
+  log_topic_id    = var.topic_id
+}
```
Comment (Lance52259) [inline docs/resources/elb_logtanks.md]: keep the script format uniform, e.g.
```
variable "xxx" {}

resource "" "test" {
  ...
}

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+func testAccElbV3LogTanksConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+data "huaweicloud_vpc_subnet" "test" {
+  name = "subnet-default"
+}
+data "huaweicloud_availability_zones" "test" {}
+resource "huaweicloud_elb_loadbalancer" "test" {
+  name            = "%[1]s"
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: Between resources, spaces should be used to separate the code blocks of different functions in the resources.
Parameter block ditto.

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+
+	return nil
+}
+
+func testAccElbV3LogTanksConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+data "huaweicloud_vpc_subnet" "test" {
+  name = "subnet-default"
+}
+data "huaweicloud_availability_zones" "test" {}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: Data source is not recommended.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go [hunk 12]
```diff
@@ -0,0 +1,144 @@
+package elb
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/chnsz/golangsdk/openstack/elb/v3/logtanks"
```
Comment (Lance52259) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: `chnsz` is the self-developed repository, followed by `hashicorp`.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+		DeleteContext: resourceLogTanksV3Delete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
```
Comment (Lance52259) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: Timeout block is never used.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+			},
+		},
+	}
+}
+
+func resourceLogTanksV3Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	elbClient, err := config.ElbV3Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating elb client: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: s/elb/ELB v3
ELB is an abbreviation.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+	"github.com/chnsz/golangsdk/openstack/elb/v3/logtanks"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func ResourceLogTanksV3() *schema.Resource {
```
Comment (Lance52259) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: please remove version part.

File: docs/resources/elb_logtanks.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/elb_logtanks.md]: it is not used and has been deleted

File: docs/resources/elb_logtanks.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/elb_logtanks.md]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: done

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: done

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go (hunk 7 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: done

Comment (houpeng80) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: done

File: docs/resources/elb_logtanks.md (hunk 8 above)
Comment (houpeng80) [reply docs/resources/elb_logtanks.md]: done

File: docs/resources/elb_logtanks.md (hunk 9 above)
Comment (houpeng80) [reply docs/resources/elb_logtanks.md]: done

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go (hunk 10 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 13 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: it has been deleted

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 15 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: All other files use this format, maybe it's better to keep consistent here.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 14 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: All other files use this format, maybe it's better to keep consistent here.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 12 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: this function is still developed in this package, so we still need to rely on this package

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go (hunk 6 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: done

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go (hunk 11 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 12 above)
Comment (Lance52259) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: sorry, I mean these packages needs to reorder.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 14 above)
Comment (Lance52259) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: uppercase for service name is better.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 15 above)
Comment (Lance52259) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: Those are the old functions. If the client version changes in the future, the function name will be inconsistent.

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 12 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 14 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 15 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: I think it can be treated as a later optimization and handled uniformly

**Comment by chengxiangdong** [comment]
pls rebase your code

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go (hunk 15 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_logtanks.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtank.go [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+	logStreamClient, err := config.LtsV2Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating LTS client: %s", err)
+	}
+	logGroupId := d.Get("log_group_id").(string)
+	logTopicId := d.Get("log_topic_id").(string)
+	streams, err := logstreams.List(logStreamClient, logGroupId).Extract()
+	if err != nil {
+		if apiError, ok := err.(golangsdk.ErrDefault400); ok {
+			// "LTS.0201" indicates the log group is not exist
```
Comment (hmbig2) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtank.go:163]: check this error is redundant, and return the error directly

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_logtanks_test.go]: no change !! done?

**Pull request closed without merging**